classes/class3/homework/routes/cats.js

```diff
@@ -0,0 +1,90 @@
+var path = require('path');
+var Cat = require(path.join(__dirname, '../models/cat'));
+
+var colors = ['yellow',
+              'green',
+              'red',
+              'black',
+              'blue',
+              'orange',
+              'tabby',
+              'NaN',
+              'brown',
+              'teal',
+              'lemmony yellow',
+              'grey',
+              'gray',
+              'maroon',
+              'macaroon'];
+
+var names = ['George',
+             'Malyssa',
+             'Frederick',
+             'Johan',
+             'Suzy',
+             'Mary',
+             'Skags',
+             'Robert',
+             'Tinkerbell',
+             'Mr. Whiskers',
+             'King Fluffybutt']
+
+var cats = {};
+
+function all(req, res) {
+  var colorFilter;
+  Cat.find({colors: colorFilter}, function(err, cats) {
+    res.render('cats', {cats: cats});
+  });
+}
+
+function create(req, res) {
+  var color = colors[Math.floor(Math.random() * colors.length)];
+  var name = names[Math.floor(Math.random() * names.length)];
+  var age = Math.floor(Math.random() * name.length);
+  var catObject = {
+    name: name,
+    age: age,
+    colors: color
```

Comment:
`colors` is expecting 1) an array and 2) to be called `color`, see
```js
var catSchema = mongoose.Schema({
  name: String,
  age: Number,
  color: [String]
});
```
in `cat.js`.

What you're doing will, in MongoDB, attempt to create a new attribute in the cat called `colors`, but Mongoose is preventing that from happening with it's predefined schema.